File: lib/domain/public_plan/dtos/public_plan_create_no_crops_view_context.rb
# frozen_string_literal: true

module Domain
  module PublicPlan
    module Dtos
      # 公開プラン作成で作物未選択時、select_crop 再描画に必要な文脈（HTML Presenter は表示整形のみ）。
      class PublicPlanCreateNoCropsViewContext
        attr_reader :farm, :farm_size, :crops

        def initialize(farm:, farm_size:, crops:)
          @farm = farm
          @farm_size = farm_size
          @crops = crops
        end
      end
    end
  end
end

File: crates/agrr-domain/src/public_plan/dtos/public_plan_create_no_crops_view_context.rs
//! Ruby: `Domain::PublicPlan::Dtos::PublicPlanCreateNoCropsViewContext`

use crate::public_plan::catalog::FarmSizeRecord;

/// Minimal farm for no-crops view context.
#[derive(Debug, Clone, PartialEq)]
pub struct PublicPlanFarm {
    pub id: i64,
    pub name: String,
    pub region: String,
}

/// Minimal crop row for no-crops view context.
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct PublicPlanCrop {
    pub id: i64,
    pub name: String,
}

/// Ruby: `Domain::PublicPlan::Dtos::PublicPlanCreateNoCropsViewContext`
#[derive(Debug, Clone, PartialEq)]
pub struct PublicPlanCreateNoCropsViewContext {
    pub farm: PublicPlanFarm,
    pub farm_size: FarmSizeRecord,
    pub crops: Vec<PublicPlanCrop>,
}
